db/system_distributed_keyspace.cc

```diff
@@ -96,8 +97,27 @@ schema_ptr cdc_desc() {
         return schema_builder(system_distributed_keyspace::NAME, system_distributed_keyspace::CDC_DESC, {id})
                 /* The timestamp of this CDC generation. */
                 .with_column("time", timestamp_type, column_kind::partition_key)
-                /* The set of stream identifiers used in this CDC generation. */
+                /* For convenience, the list of stream IDs in this generation is split into token ranges
+                 * which the stream IDs were mapped to (by the partitioner) when the generation was created.  */
+                .with_column("range_end", long_type, column_kind::clustering_key)
+                /* The set of stream identifiers used in this CDC generation for the token range
+                 * ending on `range_end`. */
                 .with_column("streams", cdc_streams_set_type)
+                .with_version(system_keyspace::generate_schema_version(id))
+                .build();
+    }();
+    return schema;
+}
+
+/* A user-facing table providing CDC generation timestamps. */
+schema_ptr cdc_timestamps() {
+    thread_local auto schema = [] {
+        auto id = generate_legacy_id(system_distributed_keyspace::NAME, system_distributed_keyspace::CDC_TIMESTAMPS);
+        return schema_builder(system_distributed_keyspace::NAME, system_distributed_keyspace::CDC_TIMESTAMPS, {id})
+                /* This is a single-partition table. The partition key is always "timestamps". */
+                .with_column("key", utf8_type, column_kind::partition_key)
+                /* The timestamp of this CDC generation. */
+                .with_column("time", timestamp_type, column_kind::clustering_key)
```

Comment:
Sort it descending? So you don't have to scan everything if you have some of the rows already?

Reply:
What additional utility does sorting it descending give? One can just query `time > X` if they want generations only newer than `X` (that's what clients want to know usually)

Reply:
Well, time > X can be slightly more efficient with descending sort, though in many cases it will be the same.

If there is no promoted index, then time > X with descending time is just reading rows from the beginning of the partition. With ascending time, you read and discard rows until you reach time > X.

If there is a promoted index, we'll seek it in both cases.

Reply:
with descending order we can fetch the latest generation with LIMIT 1